query.py

```diff
@@ -626,13 +678,17 @@ def execute_from_tree(self) -> bool:
                 # Note that passing query=self here means that the
                 # "query" field of the TreeStateDict is populated,
                 # turning it into a QueryStateDict.
-                if self._tree.process_queries(self, self._session, processor):
+                if self._tree.process_queries(
+                    self,
+                    self._session,
+                    processor,
+                ):
                     # This processor found an answer, which is already stored
                     # in the Query object: return True
                     return True
             except Exception as e:
                 logging.error(
-                    f"Exception in execute_from_tree('{processor.__name__}') "
+                    f"Exception in execute_from_tree('{processor['__name__']}') "
```

Comment:
Is `__name__` definitely in the dictionary? (Despite starting with an underscore.)

Reply:
Yes, `__name__` should always be imported from the tree processors (the underscore attributes are only ignored in the `util` modules. I'm gonna change this to a `.get` call though, just to be 100% sure we don't get an exception while logging another exception.